server/routerlicious/packages/services-utils/src/morganLoggerMiddleware.ts

```diff
@@ -34,6 +34,24 @@ export function alternativeMorganLoggerMiddleware(loggerFormat: string) {
 	return morgan(loggerFormat, { stream });
 }
 
+/**
+ * @internal
+ */
+interface IResponseLatency {
+	/**
+	 * Emitted when the response has been sent.
+	 * More specifically, this event is emitted when the last segment of the response headers and body have been handed off to the operating system for transmission over the network.
+	 * It does not imply that the client has received anything yet.
+	 * Docs: https://nodejs.org/docs/latest-v16.x/api/http.html#event-finish_1
+	 */
+	finishTime: number | undefined;
+	/**
+	 * Indicates that the response is completed, or its underlying connection was terminated prematurely (before the response completion).
+	 * Docs: https://nodejs.org/docs/latest-v16.x/api/http.html#event-close_2
+	 */
+	closeTime: number;
```

Comment:
Can you please help me validate my understanding:

When the request is handled successfully and there is no timeout: both close and finish will be called with finish having a very small time delay compared to when close is emitted or does finish come before close?

If it is prematurely terminated, only close will be called.

If the request errored out but it is not the underlying connection being terminated, both finish and close will be called?

Reply:
`finish` comes before `close`, normally with a very small time difference.

Correct that if the request is prematurely ended (e.g. by client due to timeout) then only `close` will be called.

I believe, on error, `error` would happen, then `close`, with no `finish`. It's also possible that there would be `error` but nothing else, which is why I added the `reject on error` code to be safe.

Reply:
There is also an event in the parent class of ServerResponse, OutgoingMessage, called [prefinish](https://nodejs.org/docs/latest-v20.x/api/http.html#event-prefinish), which might be worth logging here.